microsoft/surface/common/kernel/linux-surface/default.nix

```diff
@@ -7,14 +7,14 @@ let
 
   cfg = config.microsoft-surface;
 
-  version = "6.10.5";
+  version = "6.11.4";
   kernelPatches = surfacePatches {
     inherit version;
     patchFn = ./patches.nix;
   };
   kernelPackages = linuxPackage {
     inherit version kernelPatches;
-    sha256 = "02yckkh6sxvcrwzbqgmw4jhqhxmbvz87xn9wm6bwwka3w2r9x41h";
+    sha256 = "0mcg1rrw9b0lwj88jkaw6ic2mks8xh8i92v90sbr2x35ljhb0m5x";
```

Comment:
I'm currently trying to bump surface-kernel to `6.12.4`.

>  `sha256 = "0mcg1rrw9b0lwj88jkaw6ic2mks8xh8i92v90sbr2x35ljhb0m5x";`

This is not the kernel checksum but the kernel+patches checksum or?
Can someone please explain how to get this sha256?


When just bumping version to `6.12.4` I get:
```
linux> trying https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.12.4.tar.xz
linux>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
linux>                                  Dload  Upload   Total   Spent    Left  Speed
linux> 100  141M  100  141M    0     0  11.0M      0  0:00:12  0:00:12 --:--:-- 11.1M
error: hash mismatch in fixed-output derivation '/nix/store/b3kgpwhzrqwi96nk0l1rmk2lsai0q36i-linux-6.12.4.tar.xz.drv':
         specified: sha256-vVSwoKRldJGXBmmLFBHsSM8qWDRcTYmQ5BSsxHMOj1U=
            got:    sha256-bzX4IUM9hCG+cWeZB0fHxKDEUZWPuWiDRGMBrxPXEVI=
```

Looks like `sha256-vVSwoK...` is the sha256 from kernel 6.11.4
```bash
nix store prefetch-file https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.11.4.tar.xz
Downloaded 'https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.11.4.tar.xz' to '/nix/store/rzyyh0ah0praswrqiql36cwpi3c976qa-linux-6.11.4.tar.xz' (hash 'sha256-vVSwoKRldJGXBmmLFBHsSM8qWDRcTYmQ5BSsxHMOj1U=').
```

and `sha256-bzX4IUM..` is the sha256 from kernel 6.12.4

```bash
nix store prefetch-file https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.12.4.tar.xz
Downloaded 'https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.12.4.tar.xz' to '/nix/store/3xn609vfg4pzxg3ip5nkfqgb6bw2iihm-linux-6.12.4.tar.xz' (hash 'sha256-bzX4IUM9hCG+cWeZB0fHxKDEUZWPuWiDRGMBrxPXEVI=').
```

Reply:
I just saw this, but it's extremely late in my locale so I plan to reply tomorrow.

One thing, is it might be easier to follow-up your message by either opening a new issue or PR with the details, and CC the people you want help from.

This can be myself, if it's for Microsoft Surface (among a few other devices) or probably any others who've recently contributed to the code under `//microsoft/surface/`.

Reply:
Thank you very much for your quick reply. I still can't reproduce the checksum, but I was still able to build a new kernel (see PR https://github.com/NixOS/nixos-hardware/pull/1212 ), so it's ok for me.